simulator/autoware_auto_msgs_adapter/README.md

```diff
@@ -0,0 +1,96 @@
+- Multiple instances can be launched to convert multiple message types.
+- Can be launched as a standalone node or as a component.
+
+## Usage
+
+Customize the adapter configuration by replicating and editing the `adapter_control.param.yaml` file located
+in the `autoware_auto_msgs_adapter/config` directory. Example configuration:
+
+```yaml
+/**:
+  ros__parameters:
+    msg_type_target: "autoware_auto_control_msgs::msg::AckermannControlCommand"
+    topic_name_source: "/control/command/control_cmd"
+    topic_name_target: "/control/command/control_cmd_auto"
+```
+
+Set the `msg_type_target` parameter to the desired target message type from `autoware_auto_msgs`.
+
+Make sure that the `msg_type_target` has the correspondence
+in [src/autoware_auto_msgs_adapter_core.cpp](src/autoware_auto_msgs_adapter_core.cpp).
+
+Launch the adapter node by any of the following methods:
+
+### `ros2 launch`
+
+```bash
+ros2 launch autoware_auto_msgs_adapter autoware_auto_msgs_adapter.launch.xml param_path:='full_path_to_param_file'
+```
+
+Make sure to set the `param_path` argument to the full path of the parameter file.
+
+Alternatively,
+
+- You can replicate and edit the launch file to suit to your needs.
+- You can make use of the existing launch file in another launch file by providing the parameter file path as an
+  argument.
+
+### `ros2 run`
+
+```bash
+ros2 run autoware_auto_msgs_adapter autoware_auto_msgs_adapter_exe --ros-args --params-file 'full_path_to_param_file'
+```
+
+Make sure to set the `param_path` argument to the full path of the parameter file.
+
+## Contributing
+
+### Current implementation details
+
+The entry point for the adapter executable is created with `RCLCPP_COMPONENTS_REGISTER_NODE` the [autoware_auto_msgs_adapter_core.cpp](src/Fautoware_auto_msgs_adapter_core.cpp).
+
+This allows it to be launched as a component or as a standalone node.
+
+In the `AutowareAutoMsgsAdapterNode` constructor, the adapter is selected by the type string provided in the
+configuration file. The adapter is then initialized with the topic names provided.
+
+The constructors of the adapters are responsible for creating the publisher and subscriber (which makes use of the conversion method).
+
+### Adding a new message pair
+
+To add a new message pair,
+
+- Replicate and edit:
+  - [adapter_control.hpp](include/autoware_auto_msgs_adapter/adapter_control.hpp).
+  - Add the new header file to the [CMakeLists.txt](CMakeLists.txt).
+- Add a new entry in the if-else if block in the constructor of the adapter node:
+  - [autoware_auto_msgs_adapter_core.cpp](src/autoware_auto_msgs_adapter_core.cpp)
+- Create a new config file by replicating and editing:
+  - [adapter_control.param.yaml](config/adapter_control.param.yaml)
+- Add a new test file by replicating and editing:
+  - [test_msg_ackerman_control_command.cpp](test/test_msg_ackerman_control_command.cpp)
```

Comment:
spell-check-differential
```suggestion
  - [test_msg_ackermann_control_command.cpp](test/test_msg_ackerman_control_command.cpp)
```

Reply:
My mistake, fixed in: https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/66edca24d3028a10fe4eae95008f80538f14d0f0